cfpb/consumerfinance.gov
# Simplify HeadingBlock editor implementation

Our HeadingBlock block has some customization of its editing interface in the Wagtail admin. Instead of its sub-blocks (text, heading level, and icon) displayed below each other, they are displayed side-by-side.

Currently this is done by providing a custom editing template for the block (struct-with-block-wrapper-classes.html) which is configured via the block's form_template property, documented [here](https://docs.wagtail.org/en/latest/advanced_topics/customisation/streamfield_blocks.html#custom-editing-interfaces-for-structblock). We also provide custom CSS which targets the classes introduced via the custom template.

Over time, we haven't kept this template up to date with changes in the upstream Wagtail editor template. Additionally, this approach requires that we define custom sub-block types (HeadingTextBlock, HeadingLevelBlock, HeadingIconBlock) even though these are functionally equivalent to Wagtail build-in block types (CharBlock and ChioceBlock).

This commit deprecates the custom template in favor of targeting the single block classname and its children, via their data-contentpath attribute (as recommended and documented by Wagtail in the link above). This also allows us to deprecate those custom sub-block types. This requires us to do some surgery on existing migration files that refer to them; because they are functionally equivalent, we don't need to create a new migration.

## How to test this PR

To test, edit a page with a HeadingBlock (for example, an InfoUnitGroup or Table) and verify that the editor interface looks the same.

## Screenshots

|Before|After|
|-|-|
|![image](https://github.com/cfpb/consumerfinance.gov/assets/654645/510eb0f2-82fb-4cb3-92ce-a84db2c1856f)|![image](https://github.com/cfpb/consumerfinance.gov/assets/654645/3cd98a4c-3c2d-442b-a61c-275354b79c2a)|

Note that there is a slight difference, in that with this change we get Wagtail's default label styling (which looks a little nicer to my eye). We also get back the * indicating a required input, which was previously removed; we don't remove this generally so I think it's better to remove the complexity.

## Checklist
